test/e2e/cert_rotation_test.go

```diff
@@ -61,17 +58,28 @@ func certRotation() {
 		"mariadb-cert",
 	}
 	for _, secretName := range secretList {
-		err := targetClusterClientSet.CoreV1().Secrets(ironicNamespace).Delete(ctx, secretName, metav1.DeleteOptions{})
+		err := clientSet.CoreV1().Secrets(ironicNamespace).Delete(ctx, secretName, metav1.DeleteOptions{})
 		Expect(err).To(BeNil(), "Cannot detele this secret: %s", secretName)
 	}
 
 	By("Wait for containers in the ironic pod to be restarted")
 	Eventually(func() error {
-		ironicPod, err := getPodFromDeployment(targetCluster, ironicDeployment, ironicNamespace)
+		ironicPod, err := getPodFromDeployment(clientSet, ironicDeployment, ironicNamespace)
 		if err != nil {
 			return err
 		}
-
+		for container := range containerNumRestart {
+			notFound := true
+			for _, ironicContainer := range ironicPod.Status.ContainerStatuses {
+				if ironicContainer.Name == container {
+					notFound = false
+					break
+				}
+			}
+			if notFound {
+				return fmt.Errorf("%s container does not exist in Ironic pod", container)
+			}
+		}
```

Comment:
I do not think we need this since already included in the next loop we can check `exist` than return `error` if `false`

Reply:
True that's different check then